processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -1024,17 +1027,20 @@ public QueryableIndex load(File inDir, ObjectMapper mapper) throws IOException
         segmentBitmapSerdeFactory = new BitmapSerde.LegacyBitmapSerdeFactory();
       }
 
-      Map<String, Object> metadata = null;
+      Metadata metadata = null;
       ByteBuffer metadataBB = smooshedFiles.mapFile("metadata.drd");
       if (metadataBB != null) {
         try {
```

Comment:
there is no `unmap` for "files" inside the smooshed file, whole segment file would get `unmapped` when closed.